src/Nethermind/Nethermind.Blockchain/Processing/BlockchainProcessor.cs

```diff
@@ -377,6 +315,68 @@ public Block Process(Block suggestedBlock, ProcessingOptions options, IBlockTrac
             return lastProcessed;
         }
 
+        private void TraceFailingBranch(ProcessingBranch processingBranch, IBlockTracer blockTracer, ProcessingOptions options)
```

Comment:
Can we make it generic based on BlockTracerBase ? So we don't have to do casts in LogDiagnosticTrace